controllers/user.controllers.js

```diff
@@ -72,28 +71,25 @@ module.exports.loginAction = async (req, res, next) => {
 
 //rendering to user home page after successful login
 module.exports.home = (req, res, next) => {
-    res.render('userHome',{ uid:req.session.user })
+    res.render('userHome',{ uid:req.session.user.email })
 }
 
 //function to get user profile who have loggged in
-module.exports.profile = async (req, res, next) => {
+module.exports.profile = (req, res, next) => {
     try{
-        const resultUser = await user.findOne({email:req.session.user});
-            if(resultUser != null) {
-                res.render('userProfile',{userData:resultUser, uid:req.session.user, msg:req.flash('msg'), msg1:req.flash('msg1')})
-            }
+        const resultUser = req.session.user;
+        res.render('userProfile',{userData:resultUser, uid:req.session.user.email, msg:req.flash('msg'), msg1:req.flash('msg1')})
     } catch(err) {
         console.log(err);
     }
 }
 
 //function for rendering on edit profile page with user data
-module.exports.editProfile = async (req, res, next) => {
+module.exports.editProfile = (req, res, next) => {
     try{
-        const resultUser = await user.findOne({email:req.session.user});
-            if(resultUser != null) {
-                res.render('updateProfile',{ data:resultUser, uid:req.session.user} )
-            }
+        const resultUser = req.session.user;
+        res.render('updateProfile',{ data:resultUser, uid:req.session.user.email, msg1:req.flash('msg1')} )
```

Comment:
uid:req.session.user.email, should be uid:resultUser.email, as session data already in resultUser